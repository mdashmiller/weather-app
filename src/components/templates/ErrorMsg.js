import React from 'react'

const ErrorMsg = () =>
	<div>
		<h2>It looks like openweathermap.org is having some issues.</h2>
		<h2>Please try again in a few minutes.</h2>
	</div>

export default ErrorMsg
