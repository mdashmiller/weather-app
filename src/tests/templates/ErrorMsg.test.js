import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import ErrorMsg from '../../components/templates/ErrorMsg'

Enzyme.configure({ adapter: new Adapter() })

describe('<ErrorMsg /> rendering', () => {

	it('renders without crashing', () => {
	  const div = document.createElement('div')
	  ReactDOM.render(<ErrorMsg />, div)
	  ReactDOM.unmountComponentAtNode(div)
	})

	test('has a valid snapshot', () => {
		const component = renderer.create(<ErrorMsg />)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('renders 1 div', () => {
		const wrapper = shallow(<ErrorMsg />)
		expect(wrapper.find('div').length).toBe(1)
	})

	it('renders 2 h2', () => {
		const wrapper = shallow(<ErrorMsg />)
		expect(wrapper.find('h2').length).toBe(2)
	})	

})
