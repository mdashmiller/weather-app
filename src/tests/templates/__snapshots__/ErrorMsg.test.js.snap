// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMsg /> rendering has a valid snapshot 1`] = `
<div>
  <h2>
    It looks like openweathermap.org is having some issues.
  </h2>
  <h2>
    Please try again in a few minutes.
  </h2>
</div>
`;
